### PR TITLE
[operator] fix multi status update in reconcile()

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

* fix `go.mod`
* Only update status if needed in the main reconcile() fonction
* Refactor reconcile() sub fonction call
* An Operator logs streaming in e2e test
* Improve `make validate` output

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,7 +58,7 @@ clean:
 
 validate: bin/golangci-lint bin/wwhrd
 	./bin/golangci-lint run ./...
-	./hack/verify-license.sh
+	./hack/verify-license.sh > /dev/null
```
Comment (clamoriniere) [inline Makefile]: redirect only stdout, we are looking only to the stderr in this case

File: pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go [hunk 2]
```diff
@@ -203,35 +203,29 @@ func (r *ReconcileDatadogAgentDeployment) Reconcile(request reconcile.Request) (
...
+		}
 	}
 
-	result, err = r.reconcileAgent(reqLogger, instance, newStatus)
-	if err != nil {
-		reqLogger.Error(err, "unable to reconcile the Agent")
-		return r.updateStatusIfNeeded(reqLogger, instance, newStatus, result, err)
+	// Always requeue
+	if !result.Requeue {
+		result.RequeueAfter = 15 * time.Second
```
Comment (ahmed-mez) [inline pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go]: could we define at as const?

File: pkg/controller/datadogagentdeployment/datadogagentdeployment_controller_test.go [hunk 3]
```diff
@@ -297,18 +300,11 @@ func TestReconcileDatadogAgentDeployment_Reconcile(t *testing.T) {
...
-					version := getAgentVersion(dad)
-					_ = c.Create(context.TODO(), buildAgentClusterRole(dad, resourceName, version))
-					_ = c.Create(context.TODO(), buildClusterRoleBinding(dad, roleBindingInfo{
-						name:               resourceName,
-						roleName:           resourceName,
-						serviceAccountName: resourceName,
-					}, version))
-					_ = c.Create(context.TODO(), buildServiceAccount(dad, resourceName, version))
+
+					createAgentDependencies(c, dad)
```
Comment (ahmed-mez) [inline pkg/controller/datadogagentdeployment/datadogagentdeployment_controller_test.go]: 👍

File: test/e2e/datadogagentdeployment_test.go [hunk 4]
```diff
@@ -8,29 +8,30 @@ package e2e
...
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	dynclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
 	retryInterval        = time.Second * 5
-	timeout              = time.Second * 60
+	timeout              = time.Second * 120
```
Comment (ahmed-mez) [inline test/e2e/datadogagentdeployment_test.go]: I believe this change was for testing purpose right?

File: test/e2e/utils/new.go [hunk 5]
```diff
@@ -38,8 +39,15 @@ func NewDatadogAgentDeployment(ns, name, image string, options *NewDatadogAgentD
...
 		Agent: &datadoghqv1alpha1.DatadogAgentDeploymentSpecAgentSpec{
-			Image:              datadoghqv1alpha1.ImageConfig{},
-			Config:             datadoghqv1alpha1.NodeAgentConfig{},
+			Image: datadoghqv1alpha1.ImageConfig{},
+			Config: datadoghqv1alpha1.NodeAgentConfig{
+				Resources: &v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("0"),
+						v1.ResourceMemory: resource.MustParse("0"),
+					},
```
Comment (ahmed-mez) [inline test/e2e/utils/new.go]: not sure if we want to keep these too

File: test/e2e/datadogagentdeployment_test.go (hunk 4 above)
Comment (clamoriniere) [reply test/e2e/datadogagentdeployment_test.go]: in some case 120 second maybe require like: rolling update of a new daemonset.
120sec seems not so high

File: test/e2e/utils/new.go (hunk 5 above)
Comment (clamoriniere) [reply test/e2e/utils/new.go]: it will help de schedule the daemonset on low test cluster

**Pull request merged**